src/utils/transform-filters.ts

```diff
@@ -1,60 +1,70 @@
  */
-function combineFiltersWithOR(filters: Filter[]): Filter | CombinedFilter {
+function combineFiltersWithOR(filters: FieldValueStaticFilter[]): StaticFilter {
   if (filters.length === 1) {
     return filters[0];
   }
   return {
+    kind: 'disjunction',
     combinator: FilterCombinator.OR,
-    filters: filters
+    filters
   };
 }
 
 /**
- * Converts a list of {@link SelectableFilter}s used in Search Headless
- * to a single nested filter stucture used in Search Core.
+ * Converts a list of {@link SelectableStaticFilter}s used in Search Headless
+ * to a single static filter expected by Search Core.
  *
  * @param selectableFilters - The filters to be transformed
- * @returns The filters in a singly-nested {@link CombinedFilter}, or if there
- *          is only one filter in the list and it is selected, returns that
- *          {@link Filter}
+ * @returns The filters combined into a single {@link StaticFilter}
  */
 export function transformFiltersToCoreFormat(
-  selectableFilters: SelectableFilter[] | undefined
-): Filter | CombinedFilter | null {
+  selectableFilters: SelectableStaticFilter[] | undefined
+): StaticFilter | null {
   if (!selectableFilters) {
     return null;
   }
-  if (selectableFilters.length === 0) {
+
+  const selectedFilters: StaticFilter[] = selectableFilters
+    .filter(selectableFilter => selectableFilter.selected)
+    .map(selectableFilter => selectableFilter.filter);
+  if (selectedFilters.length === 0) {
     return null;
   }
-  if (selectableFilters.length === 1) {
-    const { selected, displayName: _, ...filter } = selectableFilters[0];
-    return selected ? filter : null;
+  if (selectedFilters.length === 1) {
+    return selectedFilters[0];
   }
-  const selectedFilters = selectableFilters.filter(selectableFilter => selectableFilter.selected);
-  const groupedFilters: Record<string, Filter[]> = selectedFilters.reduce((groups, element) => {
-    const { selected: _, displayName: __, ...filter } = element;
-    groups[filter.fieldId]
-      ? groups[filter.fieldId].push(filter)
-      : groups[filter.fieldId] = [filter];
-    return groups;
-  }, {});
 
-  const groupedFilterLabels = Object.keys(groupedFilters);
-  if (groupedFilterLabels.length === 1) {
-    return combineFiltersWithOR(groupedFilters[groupedFilterLabels[0]]);
+  const combinationFilters: StaticFilter[] = [];
+  const fieldIdToFilters: Record<string, FieldValueStaticFilter[]> = selectedFilters.reduce(
+    (fieldIdToFilters, filter) => {
+      if (filter.kind !== 'fieldValue') {
+        combinationFilters.push(filter);
+      } else {
+        fieldIdToFilters[filter.fieldId]
+          ? fieldIdToFilters[filter.fieldId].push(filter)
+          : fieldIdToFilters[filter.fieldId] = [filter];
+      }
+      return fieldIdToFilters;
+    }, {}
+  );
+
+  const groupedFilterLabels = Object.keys(fieldIdToFilters);
+  if (groupedFilterLabels.length === 1 && combinationFilters.length === 0) {
```

Comment:
how about the case that we only have 1 item in combinationFilters, we should just return that without making a conjunction right?

Reply:
this case is handled by line 41 where there is only 1 selected filter. the reason a special check is needed for the grouped filters is because those could have started as multiple selected filters, but then been grouped into a single filter combined with OR